docs/design-documents/20241024-conduit-cli.md

```diff
@@ -0,0 +1,412 @@
+- The CLI needs to feel intuitive and align with our standards.
+
+## Background
+
+Here's how the existing Conduit CLI looks like:
+
+```shell
+$ conduit --help
+Usage of conduit:
+  -api.enabled
+    	enable HTTP and gRPC API (default true)
+  -config string
+    	global config file (default "conduit.yaml")
+  -connectors.path string
+    	path to standalone connectors' directory (default "./connectors")
+  -db.badger.path string
+    	path to badger DB (default "conduit.db")
+  -db.postgres.connection-string string
+    	postgres connection string
+  -db.postgres.table string
+    	postgres table in which to store data (will be created if it does not exist) (default "conduit_kv_store")
+  -db.type string
+    	database type; accepts badger,postgres,inmemory (default "badger")
+  -grpc.address string
+    	address for serving the gRPC API (default ":8084")
+  -http.address string
+    	address for serving the HTTP API (default ":8080")
+  -log.format string
+    	sets the format of the logging; accepts json, cli (default "cli")
+  -log.level string
+    	sets logging level; accepts debug, info, warn, error, trace (default "info")
+  -pipelines.exit-on-error
+    	exit Conduit if a pipeline experiences an error while running
+  -pipelines.path string
+    	path to the directory that has the yaml pipeline configuration files, or a single pipeline configuration file (default "./pipelines")
+  -processors.path string
+    	path to standalone processors' directory (default "./processors")
+  -version
+    	prints current Conduit version
+```
+
+## Suggested commands for Conduit 0.13
+
+The following list contains the suggested commands we propose to include in the first iteration. New and more complex commands will be added later.
+
+<details>
+<summary><code style="font-size: 19px; font-weight:bold;">conduit init</code></summary>
+
+#### Description
+
+- This command will initialize a Conduit working environment creating the `conduit.yaml` configuration file, and the three directories: processors, pipelines, and connectors.
+- It does not require having conduit running.
+- It won't require flags or arguments.
+- Additional flags could be provided to specify the path.
+
+#### Flags
+
+|Flag name | Description | Required  | Default  |
+|---|---|---|---|
+| path  |  Where to initialize Conduit | No | `.` (current directory) |
+
+#### `--help`
+
+```bash
+$ conduit init [--path PATH]
+
+EXAMPLES
+  $ conduit init
+  $ conduit init --path ~/code/conduit-dir
+```
+
+</details>
+
+<br/>
+
+<details>
+<summary><code style="font-size: 19px; font-weight:bold;">conduit config</code></summary>
+
+#### Description
+
+- This command will output the [Conduit configuration](https://github.com/ConduitIO/conduit/blob/05dbc275a724526f02779abb47b0ecc53f711485/pkg/conduit/config.go#L34) based on the default values and the user's configured settings that Conduit would use.
```

Comment:
i.e. the effective configuration? If so, then the CLI should then be accessing a running Conduit instance, because of the command line flags, custom config files, etc.

Reply:
@hariso yup, that's the idea. Conduit should be running. Is that what you meant?

Reply:
@hariso good catch. Fixed